pandas/core/internals/managers.py

```diff
@@ -1606,15 +1604,14 @@ def _consolidate_check(self):
     def _consolidate_inplace(self):
         pass
 
-    def delete(self, item):
+    def idelete(self, indexer):
```

Comment:
Are you at a point where `indexer` can be typed?

Reply:
pretty wide, anything you can pass to `np.delete`

i was thinking it might be nice to put something in `pandas._typing` to represent valid positional indexers

Reply:
That does sound generally useful, but is out of scope for this PR.

Reply:
ahh i see, yes this is useful, but in a followon